plugins/usbdmx/ShowJockeyDMXU1.cpp

```diff
@@ -0,0 +1,362 @@
+                        LIBUSB_ENDPOINT_IN;
+        if (endpoint->bmAttributes == LIBUSB_TRANSFER_TYPE_BULK && isOutput) {
+          OLA_INFO << "Using interface " << i;
+          *interface_number = i;
+          *endpoint_addr = current_endpoint_addr;
+          *max_packet_size = endpoint->wMaxPacketSize;
+          adaptor->FreeConfigDescriptor(device_config);
+          return true;
+        }
+      }
+    }
+  }
+  OLA_WARN << "Failed to locate endpoint for ShowJockeyDMXU1 device.";
+  adaptor->FreeConfigDescriptor(device_config);
+  return false;
+}
+}  // namespace
+
+// ShowJockeyDMXU1ThreadedSender
+// -----------------------------------------------------------------------------
+
+/*
+ * Sends messages to a ShowJockeyDMXU1 device in a separate thread.
+ */
+class ShowJockeyDMXU1ThreadedSender: public ThreadedUsbSender {
+ public:
+  ShowJockeyDMXU1ThreadedSender(LibUsbAdaptor *adaptor,
+                                libusb_device *usb_device,
+                                libusb_device_handle *usb_handle,
+                                int max_packet_size_out,
+                                int endpoint);
+
+ private:
+  LibUsbAdaptor* const m_adaptor;
+  int m_max_packet_size_out;
+  int m_endpoint;
+
+  bool TransmitBuffer(libusb_device_handle *handle,
+                      const DmxBuffer &buffer);
+  int bulkSync(libusb_device_handle *handle,
+               int endpoint,
+               int max_packet_size,
+               unsigned char *buffer,
+               int size);
+};
+
+ShowJockeyDMXU1ThreadedSender::ShowJockeyDMXU1ThreadedSender(
+    LibUsbAdaptor *adaptor,
+    libusb_device *usb_device,
+    libusb_device_handle *usb_handle,
+    int max_packet_size_out,
+    int endpoint)
+    : ThreadedUsbSender(usb_device, usb_handle),
+      m_adaptor(adaptor),
+      m_max_packet_size_out(max_packet_size_out),
+      m_endpoint(endpoint) {
+}
+
+bool ShowJockeyDMXU1ThreadedSender::TransmitBuffer(libusb_device_handle *handle,
+                                                   const DmxBuffer &buffer) {
+  if (!handle) {
+    return false;
+  }
+
+  int ret_val = 0;
+  int left_write_size = DMX_UNIVERSE_SIZE;
+  uint16_t already_written_size = 0;
+  unsigned int write_size = 0;
+  int max_packet_size_out = m_max_packet_size_out;
+
+  unsigned char *bulk_buffer = new unsigned char[max_packet_size_out]();
+
+  int bulk_data_size = max_packet_size_out - 2;
+  unsigned int slot = 0;
+
+  while (left_write_size > 0) {
+    memset(bulk_buffer, 0, max_packet_size_out);
+    ola::utils::SplitUInt16(already_written_size,
+                            &bulk_buffer[1],
+                            &bulk_buffer[0]);
+    write_size = std::min(bulk_data_size, left_write_size);
+
+    buffer.GetRange(slot, bulk_buffer + 2, &write_size);
+
+    ret_val = bulkSync(handle, m_endpoint, max_packet_size_out,
+                       bulk_buffer, write_size + 2);
+
+    if (ret_val < 0) {
+      return ret_val;
```

Comment:
You have a potential leak here - of this returns bulk_buffer will leak.

To avoid this make bulk_buffer a member of the ShowJockeyDMXU1ThreadedSender class. That'll also avoid allocating new memory for each frame.